### PR TITLE
UPTM-14 - Dashboard Display on vertical 4K screen

### PR DESCRIPTION
**Pull request by gitstart**
Here is the GitStart Ticket for this pull request: [UPTM-13](https://developers.gitstart.com/client/Uptime-Kuma/ticket/UPTM-13)

- Please mark your PR as "ready for review" when you have completed all requirements.
- Please ensure that your PR is reviewed by at least 2 people after you have completed it.
- Comments for review or requested changes from the upstream client will be notified to you by email and will also show on the [ticket page](https://developers.gitstart.com/client/Uptime-Kuma/ticket/UPTM-13).
- After the PR is merged on the upstream, please merge this PR.
- If there exist merge conflicts on your current PR and you have to close the PR instead of merging it, please go to the [ticket page](https://developers.gitstart.com/client/Uptime-Kuma/ticket/UPTM-13) to manually finish the task that is related to this PR.

### Loom video
https://www.loom.com/share/ef13e3e6a9b34f7d84f568ec1d1fdd6c
Happy hacking! 🎉

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
I cannot reproduce the fix.

Is my testing environment off?
My current testing is that I have this page open (slightly zoomed out on a 5k screen):
image
My understanding, was that reloading the page should show more heartbeats, right?

I have also tried this reproduction step:
image

How can this PR be tested?

**Comment by CommanderStorm** [comment]
(if possible)
Could you add the following to the description? (This will make merging this PR automatically close the linked issue)
`Resolves https://github.com/louislam/uptime-kuma/issues/2993`

**Comment by louislam** [comment]
I tested it on my 4k tv, it is OK, need to refresh the page though.

File: src/pages/DashboardHome.vue [hunk 1]
```diff
@@ -134,6 +140,18 @@ export default {
...
     },
+    watch: {
+        importantHeartBeatList() {
+            this.$nextTick(() => {
+                const tableContainerHeight = this.$refs.tableContainer.offsetHeight;
+                const availableHeight = this.calculatedHeight - tableContainerHeight;
+                const additionalPerPage = Math.floor(availableHeight / 58);
+                if (additionalPerPage > 0) {
+                    this.perPage = Math.max(25, this.perPage + additionalPerPage);
+                }
```
Comment (louislam) [inline src/pages/DashboardHome.vue:151]: Should we change it to the window resize event rather than inside here?

**Pull request merged**